File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        res = [-1,-1]
        i = 0
        while i < len(nums):
            if (target - nums[i]) in hashmap:
                res[0] = i
                res[1] = hashmap.get(target - nums[i])
                return res
            hashmap[nums[i]] = i
            i = i + 1 
        return res




        

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hash = new HashMap<>();
        int i = 0;
        while (i < nums.length){
            int curr = target - nums[i];
            if(hash.containsKey(curr)){
                return new int[] {i, hash.get(curr)};
            }
            hash.put(nums[i], i);
            i++;
        }
        return new int[] {-1,-1};
                
    }
}